rector.php

```diff
@@ -155,8 +149,6 @@
         // use mt_rand instead of random_int on purpose on non-cryptographically random
         RandomFunctionRector::class,
 
-        SimplifyRegexPatternRector::class,
```

Comment:
I don't like the rule. Because it may change the behavior.
In PHP, `[0-9]` and `\d` are not exactly the same when using `u` option.
https://3v4l.org/N8sQ5

Reply:
Ok, it seems due to called on `$rectorConfig->rule()`, I removed it and revert its change on tests files https://github.com/codeigniter4/CodeIgniter4/pull/8944/commits/48f36ef37a2d915e6e007c7a089266b863b30921

Reply:
btw, the `u` usage already skipped on rector :) https://getrector.com/demo/00372e82-e4bc-451b-a7a2-495637f29d88

Reply:
Thanks!

Reply:
@kenjis do you want to re-enable `SimplifyRegexPatternRector` ? as it already skipped on `u` usage, see https://getrector.com/demo/00372e82-e4bc-451b-a7a2-495637f29d88

Reply:
I don't want. Because these patterns have a bit different meanings.

An answer by ChatGPT:
In PHP's PCRE, `\d` and `[0-9]` both match digits but are not exactly the same.

1. **Meaning**:
   - `\d` matches any digit character, including Unicode digits in some cases.
   - `[0-9]` matches only ASCII digits (0-9).

2. **Compatibility**:
   - `\d` can match non-ASCII digits if Unicode is enabled.
   - `[0-9]` always matches ASCII digits only.

3. **Performance**:
   - `[0-9]` can be slightly faster since it's a simple character class.

### Example

```php
$pattern_d = '/\d/';
$pattern_class = '/[0-9]/';
$string = "123abc";

preg_match($pattern_d, $string); // Matches
preg_match($pattern_class, $string); // Matches
```

### Conclusion

Use `[0-9]` for consistent matching of ASCII digits. Use `\d` if you need to match broader digit sets, considering encoding and Unicode support.

Reply:
By the way, where does this code https://github.com/rectorphp/rector/blob/main/rules/CodeQuality/Rector/FuncCall/SimplifyRegexPatternRector.php check for the `u` option?

Reply:
It only check delimiter, which limited to `~`, `#`, `/` via `RegexPatternDetector`,  so `u` in the last will always be skipped.